Adding a helper class for simple image processing

dotlogix opened the pull request:
Hey there I just created this simple image processing class.
It is not feature complete but could be a nice addition to the toolkit.

May need additional documentation similar to what I did for the MeshBuilder?

VaclavElias commented:
Will there be any example how to use it? Maybe, in the manual section? 🙂

dotlogix commented:
Yes I plan to add a full manual same as my last PR just don't have time I can mark this PR also as a draft for now.

VaclavElias commented:
We can merge it, no problem. It is up to you :) Thanks again for more utilities for devs!

dotlogix commented:
So I added a LOT of functionality to this class. Also added a very detailed documentation about all abilities and also added a UI example.
Took me a while to complete.

Unfortunately we can not merge this as it is dependant on my first PR to the actual stride repo currently wait's for approval.
https://github.com/stride3d/stride/pull/1972

Which adds support for image effect scissoring to support this class at all 💪

VaclavElias commented:
Thank you @dotlogix. That's ok, we can wait. I had to do also some tiny PR's to the engine a year ago, so the code only can work 🙂. Sometimes, small PR's to the engine allow big features in the toolkit 🙂

VaclavElias commented on src/Stride.CommunityToolkit/Engine/GameExtensions.cs:
```diff
@@ -64,6 +64,39 @@ async Task RootScript()
+    /// <param name="context">Optional GameContext to be used. Defaults to null.</param>
+    /// <param name="start">Optional action to execute at the start of the game. Takes the game as a parameter.</param>
+    /// <param name="update">Optional action to execute during each game loop iteration. Takes the game as a parameter.</param>
+    public static void Run(this Game game, GameContext? context = null, Action<Game>? start = null, Action<Game>? update = null)
```
Any reason for this new extension? I checked your Example06 and you should be able to do the same with the original `Run()`?

```csharp
using var game = new Game();
game.Run(start: Start);

void Start(Scene rootScene)
{
    game.SetupBase3DScene();
    var entity = game.CreatePrimitive(PrimitiveModelType.Cube);
    entity.Transform.Position = new Vector3(1f, 0.5f, 3f);
    entity.Scene = rootScene;
}
```

- `game` is already available inside the `Start` method
- `rootScene` is also available

dotlogix replied:
Ah sorry I guess I can revert that.
I mostly try to avoid closures where possible because of unnecessary display classes created by the compiler.
Mostly personal preference I guess.
The methods I allow static delegates and I much prefer this as this allows you to move the methods to a static class instead of a local function.

VaclavElias replied:
I am for any improvements. Not sure if I understand what you are saying but keep this extension there as well so I can see myself I guess 🙂

dotlogix commented:
Hey there @VaclavElias I think we can merge this now :) seems my change finally made it in a new nuget package

VaclavElias commented on src/Stride.CommunityToolkit/Engine/GameExtensions.cs:
```diff
@@ -44,8 +44,8 @@ public static class GameExtensions
         game.Script.Scheduler.Add(RootScript);
-
         game.Run(context);
+        return;
```
This is not needed?

VaclavElias commented:
Welcome back :)
